packages/core/src/utils/schema.ts

```diff
@@ -18,3 +18,52 @@ export const ChainSchema = z.custom<Chain>((chain) => {
     return false;
   }
 });
+
+const HexSchema = z.custom<`${number}px`>((val) => {
+  return typeof val === "string" ? /^0x[0-9a-fA-F]*$/.test(val) : false;
+});
+
+export const BigNumberishSchmea = z.union([HexSchema, z.number(), z.bigint()]);
+
+export const PercentageSchema = z.object({
+  /**
+   * Percent value between 1 and 1000 inclusive
+   */
+  percentage: z.number().min(1).max(1000),
+});
+
+export const UserOperationFeeOverridesFieldSchema = z.union([
+  BigNumberishSchmea,
+  PercentageSchema,
+]);
+
+export const UserOperationFeeOverridesSchema = z
+  .object({
+    /** this adds a percent buffer on top of the base fee estimated (default 50%)
+     * NOTE: this is only applied if the default fee estimator is used.
+     */
+    baseFeeBufferPercent: z.bigint().optional(),
+    /** this adds a percent buffer on top of the priority fee estimated (default 5%)'
+     * * NOTE: this is only applied if the default fee estimator is used.
+     */
+    maxPriorityFeeBufferPercent: z.bigint().optional(),
+    /** this adds a percent buffer on top of the preVerificationGas estimated
+     *
+     * Defaults 5% on Arbitrum and Optimism, 0% elsewhere
+     *
+     * This is only useful on Arbitrum and Optimism, where the preVerificationGas is
+     * dependent on the gas fee during the time of estimation. To improve chances of
+     * the UserOperation being mined, users can increase the preVerificationGas by
+     * a buffer. This buffer will always be charged, regardless of price at time of mine.
+     *
+     * NOTE: this is only applied if the defualt gas estimator is used.
+     */
+    preVerificationGasBufferPercent: z.bigint().optional(),
```

Comment:
were these part of the object before? I thought these were provider configs?

Reply:
these changes are reverted. was just a wip commit. sorry for the confusion.